base/db/R/get.new.site.R

```diff
@@ -0,0 +1,121 @@
+            # We'll try to Generate a new ID similar to previous ones.
+        } else {
+            if ((!is.null(site$lat) && !is.null(site$lon)) &&
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                new.site <- data.frame(
+                    id = as.numeric(site$id),
+                    lat = site$lat,
+                    lon = site$lon
+                )
+                str_ns <- paste0(new.site$lat, "-", new.site$lon)
+            } # siteId as well as lat/lon present
+            else {
+                new.site <- data.frame(
+                    id = site$id,
+                    lat = NULL,
+                    lon = NULL
+                )
+                str_ns <- paste0(new.site$id %/% 1e+09, "-", new.site$id %% 1e+09)
+            } # siteId present but lat/lon absent
+        }
+    } else {
+        # Check if site dataframe has an id column
+        if (is.null(site$id) | is.na(site$id)) {
+            PEcAn.logger::logger.warn("Site dataframe does not have an id column. Generating a unique ID")
+            if ((!is.null(site$lat) && !is.null(site$lon)) &&
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                PEcAn.logger::logger.info(paste0("Generated siteID using lat and lon:", site.id))
+                site.id <- generate_site_id(site$lat, site$lon)
+                new.site <- data.frame(
+                    id = as.numeric(site.id),
+                    lat = site$lat,
+                    lon = site$lon
+                )
+                str_ns <- paste0(site$lat, "-", site$lon)
+            } else {
+                PEcAn.logger::logger.severe("Missing site-id, site lat & site-lon. Stopping the process due to lack of information")
+            }
+        } else {
+            # setup site database number, lat, lon and name and copy for format.vars if new input
+            if ((!is.null(site$lat) && !is.null(site$lon)) |
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                new.site <- data.frame(
+                    id = as.numeric(site$id),
+                    lat = site$lat,
+                    lon = site$lon
+                )
+                str_ns <- paste0(site$lat, "_", site$lon)
+            }
+        }
+    }
+
+    site.info <- list(new.site = new.site, str_ns = str_ns)
+
+    return(site.info)
+}
+
+
+#' Generate a unique site ID from latitude/longitude
+#'
+#' @param lat Latitude value
+#' @param lon Longitude value
+#' @return A unique string ID generated by hashing the lat/lon values
+#' @details This function is intended for database-less runs only by generating
+#'          a unique site ID by hashing the rounded lat/lon coordinates.
+#'
+#' @author Abhinav Pandey
+#' @export
+generate_site_id <- function(lat, lon) {
+  latlon_str <- paste0(lat, lon)
+  uid <- digest::digest(latlon_str, algo = "xxhash64")
```

Comment:
Do we really need to add an entirely new package dependency to PEcAn just for this one call in one function? Is there an easier solution, or one that relies on existing packages?

Reply:
There's also `rlang::hash`